mev_inspect/aave_liquidations.py

```diff
@@ -63,26 +63,28 @@ def get_aave_liquidations(
 
 
 def _get_received_data(
-    child_traces: List[ClassifiedTrace], liquidator: str
+    liquidation: DecodedCallTrace, child_traces: List[ClassifiedTrace], liquidator: str
 ) -> Tuple[str, int]:
 
     """Look for and return liquidator payback from liquidation"""
-
     for child in child_traces:
 
         child_transfer: Optional[Transfer] = get_transfer(child)
 
-        if child_transfer is not None:
-
-            if child_transfer.to_address == liquidator:
+        if child_transfer is not None and child_transfer.to_address == liquidator:
+            return child_transfer.token_address, child_transfer.amount
 
-                return child_transfer.token_address, child_transfer.amount
+    if liquidation.error is not None:
```

Comment:
this should happen outside this function I think and just return None on the whole liquidation rather than creating dummy data

Reply:
no dummy data is created -  the data is taken from the liquidation call. In the case of Reverted liquidations, this might be useful. If not, we can return None. Why not keep the data given from the inputs for the failed liquidations?

Reply:
i think having the values as 0 is less clear than just not writing the liquidation
I've been leaning that way with swaps and arbs more too

For naive counts of liquidations / swaps etc I think it's better to take the stance of "this liquidation happened" v. "this liquidation half finished"

Reply:
Makes sense, I agree